docs/storage-usage.rst

```diff
@@ -0,0 +1,845 @@
+optional arguments
+
+.. code-block:: python
+
+   >>> blob_iterator = bucket.list_blobs(max_results=2,
+   ...                                   page_token='next-blob-name',
+   ...                                   prefix='foo',
+   ...                                   delimiter='/',
+   ...                                   versions=True,
+   ...                                   projection='noAcl',
+   ...                                   fields=None)
+   >>> for blob in blob_iterator:  # API request
+   ...     print(blob)
+
+See the `objects list`_ documentation for details.
+
+.. _objects list: https://cloud.google.com/storage/docs/json_api/v1/objects/list
+
+To delete a blob
+
+.. code-block:: python
+
+   >>> blob.delete()  # API request
+
+.. note::
+
+   You can use
+   :meth:`delete_blob() <gcloud.storage.bucket.Bucket.deltet_blob>`
+   directly to accomplish the same task.
+
+As with retrieving, you may also delete multiple blobs in a single request
+
+.. code-block:: python
+
+   >>> bucket.delete_blobs('blob-name1',
+   ...                     'blob-name2',
+   ...                     'blob-name3')  # API request
+
+This is equivalent to
+
+.. code-block:: python
+
+   >>> with client.batch():  # API request
+   ...     bucket.delete_blob('blob-name1')
+   ...     bucket.delete_blob('blob-name2')
+   ...     bucket.delete_blob('blob-name3')
+
+In the case that some of the deletes may fail, you can handle each
+error with custom behavior:
+
+.. code-block:: python
+
+   >>> def handle_not_found(blob):
+   ...     msg = '%s not found' % (blob,)
+   ...     print(msg)
+   ...
+   >>> bucket.delete_blobs('blob-name1', 'blob-name2', 'blob-name3',
+   ...                     on_error=handle_not_found)  # API request
+   blob-name2 not found
+
+To make updates to a blob use
+:meth:`patch() <gcloud.storage.blob.Blob.patch>`
+
+.. code-block:: python
+
+   >>> blob.versioning_enabled = True
+   >>> blob.patch()  # API request
+
+If there are no updates to send, an exception will occur
+
+.. code-block:: python
+
+   >>> blob.patch()  # API request
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   ValueError: No updates to send.
+
+In total, the properties that can be updated are
+
+.. code-block:: python
+
+   >>> blob.acl = [
+   ...     ACLEntity('project-owners-111111', 'OWNER'),
+   ...     ACLEntity('user-01234, 'OWNER'),
+   ... ]
+   >>> blob.cache_control = 'private, max-age=0, no-cache'
+   >>> blob.content_disposition = 'Attachment; filename=example.html'
+   >>> blob.content_encoding = 'gzip'
+   >>> blob.content_language = 'en-US'
+   >>> blob.content_type = 'text/plain'
+   >>> blob.crc32c = u'z8SuHQ=='  # crc32-c of "foo"
+   >>> blob.md5_hash = u'rL0Y20zC+Fzt72VPzMSk2A=='  # md5 of "foo"
+   >>> blob.metadata = {'foo': 'bar', 'baz': 'qux'}
+
+.. note::
+
+   **BREAKING THE FOURTH WALL**: Why are ``crc32c`` and ``md5_hash`` writable?
```

Comment:
Per the [Hashes and ETags docs](https://cloud.google.com/storage/docs/hashes-etags#_JSONAPI), they are checked (if passed) during creation of the object.